src/components/Checkbox/Checkbox.tsx

```diff
@@ -1,69 +1,77 @@
+import generateUUID from "../../helpers/generateUUID";
 
 export interface CheckboxProps {
-  /** Additional attributes passed to the checkbox */
+  /** The checkbox's label.  This will serve as the text content for a `<label>` element if `showlabel` is true, or an "aria-label" if `showLabel` is false. */
+  labelText: string;
+  /** Additional attributes to pass to the `<input>` tag */
   attributes?: { [key: string]: any };
-
-  /** BlockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
-  blockName?: string;
-  /** Unique ID used by checkbox */
-  checkboxId?: string;
-  /** ClassName that appears in addition to "checkbox" */
-  className?: string;
-  /* For a controlled component only. If checked is not defined, the state of the uncontrolled native `input` component will be used. */
+  /** When using the Checkbox as a "controlled" form element, you can specify the Checkbox's checked state using this prop. Learn more about controlled and uncontrolled form fields: https://goshakkk.name/controlled-vs-uncontrolled-inputs-react/ */
   checked?: boolean;
-  /* For uncontrolled component only. The initial value of an uncontrolled checkbox component. */
-  defaultChecked?: boolean;
-  /* The Label that the checkbox is using. */
-  labelOptions: LabelOptions;
+  /** className you can add in addition to 'input' */
+  className?: string;
+  /** Adds the 'disabled' prop to the input when true */
+  disabled?: boolean;
+  /** Helper for modifiers array; adds 'errored' styling */
+  errored?: boolean;
+  /** ID that other components can cross reference for accessibility purposes */
+  id?: string;
   /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
   modifiers?: string[];
-  /** name of the checkbox */
+  /** The name prop indicates into which group of checkboxes this checkbox belongs.  If none is specified, 'default' will be used */
   name?: string;
   /** The action to perform on the `<input>`'s onChange function  */
   onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  /** Offers the ability to show the checkbox's label onscreen or hide it. Refer to the `labelText` property for more information. */
+  showLabel: boolean;
+  /** Populates the value of the input */
+  value?: string;
 }
 
 const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
   (props, ref?) => {
     const {
-      name,
-      attributes,
-      modifiers = [],
-      className,
-      blockName = "",
-      checkboxId,
-      labelOptions,
       checked,
-      defaultChecked,
+      className,
+      disabled,
+      errored,
+      id,
+      labelText,
+      name,
       onChange,
+      showLabel,
+      value,
     } = props;
 
-    const baseClass = "checkbox";
+    const attributes = props.attributes || {};
+    const modifiers = props.modifiers ? props.modifiers : [];
+    const checkboxID = id || generateUUID();
```

Comment:
Why not set these defaults in the destruction assignment?

Reply:
Done!  (Note:  the id stays where it is because I also want the case of "" to generate a UUID. If I define that in the destructure, it only cares that the value is truthy and so it allows "" to pass.)